### PR TITLE
ENH use pygithub for tokens

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [ ] Pushed the branch to main repo
* [ ] CI passed on the branch

### CODE REVIEW

**Comment by jaimergp** [comment]
cc @beckermr

I took your snippet from the chat and tried my best in case it helped get the ball rolling. If it's easier for you to start from scratch, please feel free to close :)

**Comment by beckermr** [comment]
Looks good. Idk why it is failing. Will finish later. Thank you!

File: conda-requirements.txt [hunk 1]
```diff
@@ -6,15 +6,15 @@ conda-build
...
+cryptography>=39
 python-dateutil
 flake8
 git
 lxml
 mock
 nose
 parameterized
-pygithub>=2  # for GHA repo disptach
+pygithub>=2.2.0  # for GHA repo disptach
```
Comment (beckermr) [inline conda-requirements.txt:17]:
```suggestion
pygithub>=2.2.0  # for GH App endpoints
```

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**